rvps/src/reference_value.rs

```diff
@@ -75,13 +76,19 @@ fn default_version() -> String {
 }
 
 impl ReferenceValue {
-    pub fn new() -> Self {
-        ReferenceValue {
+    /// Create a new `ReferenceValue`, the `expired`
+    /// field's nanosecond will be set to 0. This avoid
+    /// a rare bug that when the nanosecond of the time
+    /// is not 0, the test case will fail.
```

Comment:
I think the bug can also hit us if user calls `.set_expired(Utc.ymd(2022, 9, 22).and_hms(8, 59, 59).with_nanoseconds(123)))` -- in such case serializing and deserializing will not return the original RV.  So maybe `set_expired` should be:

```rust
    pub fn set_expired(mut self, expired: DateTime<Utc>) -> Self {
        self.expired = expired.with_nanoseconds(0);
        self
    }
```

This way we make sure that the internal `expired` field of `ReferenceValue` always has 0 nanoseconds.

(not directly related to this PR which adds the file caching...)

Reply:
You're right, thanks for catching this